### ISSUE TITLE
Broken when_some which takes iterators

### ISSUE THREAD

**Issue by AntonBikineev** [STEllAR-GROUP/hpx#1745]
Non-polymorphic version of when_some seems to be broken. This simple code gives compile-time error:
```
hpx::lcos::future<hpx::when_some_result<std::vector<int> > > r =
    hpx::when_some(3u, futures.begin(), futures.end());
```
> when_some.hpp:484:20: error: no viable conversion from 'future<typename hpx::util::decay_unwrap<vector<future<int>, allocator<future<int> > > >::type> to
      future<when_some_result<std::vector<future<int> > >>
            return lcos::make_ready_future(std::move(lazy_values));

The solution is just to wrap vectors inside make_ready_future(...) to when_some_result. But I already have it fixed in my parallel_composition_with_generic_containers branch and test for such cases. If that branch doesn't get approved, we'll need to fix it.

**Comment by hkaiser** [comment]
@AntonBikineev was this fixed by merging #1739?

**Comment by AntonBikineev** [comment]
@hkaiser yes, there is a test which covers it. The issue can be closed.
